### PR TITLE
[tests-only][full-ci]Added test for banned password to create public link

### PR DESCRIPTION
**Pull request by SagarGi**
## Description
This PR adds following scenarios (creating link shares (public) to banned password):
```feature
Scenario Outline: create a file's link share with a password that is listed in the Banned-Password-List
Scenario Outline: update a file's link share with a password that is listed in the Banned-Password-List
```

## Related Issues:
https://github.com/owncloud/ocis/issues/7968

### CODE REVIEW

File: tests/acceptance/features/apiSharingNg/linkShare.feature [hunk 1]
```diff
@@ -584,3 +584,103 @@ Feature: Create a share link for a resource
...
+
+  Scenario Outline: create a file's link share with a password that is listed in the Banned-Password-List
+    Given the config "OCIS_PASSWORD_POLICY_BANNED_PASSWORDS_LIST" has been set to path "config/drone/banned-password-list.txt"
+    And user "Alice" has uploaded file with content "other data" to "text.txt"
+    When user "Alice" creates the following link share using the Graph API:
+      | resourceType    | file             |
+      | resource        | text.txt         |
+      | space           | Personal         |
+      | permissionsRole | view             |
+      | password        | <bannedPassword> |
```
Comment (PrajwolAmatya) [inline tests/acceptance/features/apiSharingNg/linkShare.feature:597]:
```suggestion
      | password        | <banned-password> |
```

File: tests/acceptance/features/apiSharingNg/linkShare.feature [hunk 2]
```diff
@@ -584,3 +584,103 @@ Feature: Create a share link for a resource
...
+                  "unfortunately, your password is commonly used. please pick a harder-to-guess password for your safety"
+                ]
+              }
+            }
+          }
+        }
+      }
+      """
+    Examples:
+      | bannedPassword |
```
Comment (PrajwolAmatya) [inline tests/acceptance/features/apiSharingNg/linkShare.feature:630]:
```suggestion
      | banned-password |
```

File: tests/acceptance/features/apiSharingNg/linkShare.feature [hunk 3]
```diff
@@ -584,3 +584,103 @@ Feature: Create a share link for a resource
...
+      | resource        | text.txt |
+      | space           | Personal |
+      | permissionsRole | view     |
+      | password        | %public% |
+    When user "Alice" sets the following password for the last link share using the Graph API:
+      | resourceType    | file             |
+      | resource        | text.txt         |
+      | space           | Personal         |
+      | permissionsRole | view             |
+      | password        | <bannedPassword> |
```
Comment (PrajwolAmatya) [inline tests/acceptance/features/apiSharingNg/linkShare.feature:650]:
```suggestion
      | password        | <banned-password> |
```

File: tests/acceptance/features/apiSharingNg/linkShare.feature [hunk 4]
```diff
@@ -584,3 +584,103 @@ Feature: Create a share link for a resource
...
+                  "unfortunately, your password is commonly used. please pick a harder-to-guess password for your safety"
+                ]
+              }
+            }
+          }
+        }
+      }
+      """
+    Examples:
+      | bannedPassword |
```
Comment (PrajwolAmatya) [inline tests/acceptance/features/apiSharingNg/linkShare.feature:683]:
```suggestion
      | banned-password |
```

**Review by ScharfViktor — APPROVED** [review]

File: tests/acceptance/features/apiSharingNg/linkShare.feature [hunk 5]
```diff
@@ -584,3 +584,103 @@ Feature: Create a share link for a resource
...
+
+  Scenario Outline: create a file's link share with a password that is listed in the Banned-Password-List
+    Given the config "OCIS_PASSWORD_POLICY_BANNED_PASSWORDS_LIST" has been set to path "config/drone/banned-password-list.txt"
+    And user "Alice" has uploaded file with content "other data" to "text.txt"
+    When user "Alice" creates the following link share using the Graph API:
+      | resourceType    | file             |
+      | resource        | text.txt         |
+      | space           | Personal         |
+      | permissionsRole | view             |
+      | password        | <banned-password> |
```
Comment (PrajwolAmatya) [inline tests/acceptance/features/apiSharingNg/linkShare.feature:597]:
```suggestion
      | resourceType    | file              |
      | resource        | text.txt          |
      | space           | Personal          |
      | permissionsRole | view              |
      | password        | <banned-password> |
```

Comment (saw-jan) [inline tests/acceptance/features/apiSharingNg/linkShare.feature:597]:
```suggestion
      | resource-type    | file             |
      | resource        | text.txt          |
      | space           | Personal          |
      | permissionsRole | view              |
      | password        | <banned-password> |
```

File: tests/acceptance/features/apiSharingNg/linkShare.feature [hunk 6]
```diff
@@ -584,3 +584,103 @@ Feature: Create a share link for a resource
...
+      | resource        | text.txt |
+      | space           | Personal |
+      | permissionsRole | view     |
+      | password        | %public% |
+    When user "Alice" sets the following password for the last link share using the Graph API:
+      | resourceType    | file             |
+      | resource        | text.txt         |
+      | space           | Personal         |
+      | permissionsRole | view             |
+      | password        | <banned-password> |
```
Comment (PrajwolAmatya) [inline tests/acceptance/features/apiSharingNg/linkShare.feature:650]:
```suggestion
      | resourceType    | file              |
      | resource        | text.txt          |
      | space           | Personal          |
      | permissionsRole | view              |
      | password        | <banned-password> |
```

**Review by PrajwolAmatya — APPROVED** [review]
other lgtm

File: tests/acceptance/features/apiSharingNg/linkShare.feature [hunk 7]
```diff
@@ -584,3 +584,103 @@ Feature: Create a share link for a resource
...
+
+  Scenario Outline: update a file's link share with a password that is listed in the Banned-Password-List
+    Given the config "OCIS_PASSWORD_POLICY_BANNED_PASSWORDS_LIST" has been set to path "config/drone/banned-password-list.txt"
+    And user "Alice" has uploaded file with content "other data" to "text.txt"
+    And user "Alice" has created the following link share:
+      | resourceType    | file     |
+      | resource        | text.txt |
+      | space           | Personal |
+      | permissionsRole | view     |
+      | password        | %public% |
```
Comment (saw-jan) [inline tests/acceptance/features/apiSharingNg/linkShare.feature:644]:
```suggestion
      | resource-type   | file     |
      | resource        | text.txt |
      | space           | Personal |
      | permissionsRole | view     |
      | password        | %public% |
```

File: tests/acceptance/features/apiSharingNg/linkShare.feature (hunk 6 above)
Comment (saw-jan) [inline tests/acceptance/features/apiSharingNg/linkShare.feature:650]:
```suggestion
      | resource-type   | file              |
      | resource        | text.txt          |
      | space           | Personal          |
      | permissionsRole | view              |
      | password        | <banned-password> |
```

**Review by saw-jan — APPROVED** [review]

**Pull request merged**